pkg/hostagent/dns/dns.go

```diff
@@ -268,8 +270,12 @@ func (h *Handler) handleQuery(w dns.ResponseWriter, req *dns.Msg) {
 		}
 	}
 	if handled {
-		reply.Truncate(truncateSize)
-		_ = w.WriteMsg(&reply)
+		if h.truncate {
+			reply.Truncate(truncateSize)
+		}
+		if err := w.WriteMsg(&reply); err != nil {
+			logrus.Debugf("handleQuery failed writing DNS reply: %v", err)
```

Comment:
Please use `%w` when formatting `error` values. This applies to a lot of changes below as well.

Reply:
Or in these cases, use `logrus.WithError(err).Debugf(...)`.